PathPlanning/FlowField/flowfield.py

```diff
@@ -0,0 +1,214 @@
+        while curr_x is not None and curr_y is not None:
+            plt.plot(curr_x, curr_y, "b*")
+            curr_x, curr_y = self.vector_field[(curr_x, curr_y)]
+            plt.pause(0.001)
+        if show_animation:
+            plt.show()
+
+
+def main():
+    # set obstacle positions
+    obs_dict = {}
+    for i in range(51):
+        for j in range(51):
+            obs_dict[(i, j)] = 'free'
+    o_x, o_y, m_x, m_y, h_x, h_y = [], [], [], [], [], []
+
+    s_x = 5.0
+    s_y = 5.0
+    g_x = 35.0
+    g_y = 45.0
+
+    # draw outer border of maze
+    draw_vertical_line(0, 0, 50, o_x, o_y, obs_dict, 'obs')
+    draw_vertical_line(48, 0, 50, o_x, o_y, obs_dict, 'obs')
+    draw_horizontal_line(0, 0, 50, o_x, o_y, obs_dict, 'obs')
+    draw_horizontal_line(0, 48, 50, o_x, o_y, obs_dict, 'obs')
+
+    # draw inner walls
+    all_x = [10, 10, 10, 15, 20, 20, 30, 30, 35, 30, 40, 45]
+    all_y = [10, 30, 45, 20, 5, 40, 10, 40, 5, 40, 10, 25]
+    all_len = [10, 10, 5, 10, 10, 5, 20, 10, 25, 10, 35, 15]
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_vertical_line(x, y, l, o_x, o_y, obs_dict, 'obs')
+
+    all_x[:], all_y[:], all_len[:] = [], [], []
+    all_x = [35, 40, 15, 10, 45, 20, 10, 15, 25, 45, 10, 30, 10, 40]
+    all_y = [5, 10, 15, 20, 20, 25, 30, 35, 35, 35, 40, 40, 45, 45]
+    all_len = [10, 5, 10, 10, 5, 5, 10, 5, 10, 5, 10, 5, 5, 5]
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_horizontal_line(x, y, l, o_x, o_y, obs_dict, 'obs')
+
+    # draw paths that have medium difficulty (in terms of going through them)
+    all_x[:], all_y[:], all_len[:] = [], [], []
+    all_x = [10, 45]
+    all_y = [22, 20]
+    all_len = [8, 5]
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_vertical_line(x, y, l, m_x, m_y, obs_dict, 'medium')
+
+    all_x[:], all_y[:], all_len[:] = [], [], []
+    all_x = [20, 30, 42] + [47] * 5
+    all_y = [35, 30, 38] + [37 + i for i in range(2)]
+    all_len = [5, 7, 3] + [1] * 3
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_horizontal_line(x, y, l, m_x, m_y, obs_dict, 'medium')
+
+    # draw paths that have hard difficulty (in terms of going through them)
+    all_x[:], all_y[:], all_len[:] = [], [], []
+    all_x = [15, 20, 35]
+    all_y = [45, 20, 35]
+    all_len = [3, 5, 7]
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_vertical_line(x, y, l, h_x, h_y, obs_dict, 'hard')
+
+    all_x[:], all_y[:], all_len[:] = [], [], []
+    all_x = [30] + [47] * 5
+    all_y = [10] + [37 + i for i in range(2)]
+    all_len = [5] + [1] * 3
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_horizontal_line(x, y, l, h_x, h_y, obs_dict, 'hard')
+
+    plt.plot(o_x, o_y, "sr")
+    plt.plot(m_x, m_y, "sg")
+    plt.plot(h_x, h_y, "sy")
+    plt.plot(s_x, s_y, "og")
+    plt.plot(g_x, g_y, "o")
+    plt.grid(True)
+
+    flow_obj = FlowField(obs_dict, g_x, g_y, s_x, s_y, 50, 50)
+    flow_obj.find_path()
```

Comment:
What is these color (yellow, green) meaning? Could you please add description to explain it.
![SS 2020-09-27 at 13 37 36](https://user-images.githubusercontent.com/3813847/94356272-6355ee80-00c7-11eb-9885-30256d496ffd.png)